### ISSUE TITLE
Non unique enums, to string handling - issues and options

### ISSUE THREAD

**Issue by quicknir** [quicknir/wise_enum#5]
If two enums have the same value, then currently the macro will not work, solely because the to_string generated function will have a switch with multiple identical case labels, which is not allowed. Conceptually, there is no clearly correct (or even particularly reasonable) way to convert an enum to a string when two enumerators have the same value. The function could be made partial (only defined for some enums), but this solution a) isn't very good, and b) is a mess to implement.

The only option I'm currently interested in providing for people who need to define enums with repeated values is to omit the to_string function entirely from the macro. This would still allow them to take advantage of all the other useful features, and perhaps allow them to build their own reasonable to_string implementation by e.g. taking advantage of the constexpr array of enums. 

Another issue worth mentioning is whether it's worth providing a more convenient way to give users an ADL enabled `to_string` function. The currently generated function is called `wise_enum_to_string`; it's an implementation detail but pollutes user namespace more than it should; it's signature should be changed to take both an enum type, and the implementation detail Tag type, to lessen the pollution. 

In short then, there are perhaps 3 "levels" of to_string'iness that the library could provide: 

 1. none (for non unique enums, or perhaps for those paranoid about code bloat and not planning to use it), 
 2. The current level. Only `wise_enum::to_string(e)` is supported, everything else is an implementation detail.
 3. ADL level. This is a pure convenience as the user can easily implement it themselves, but it would also define a `to_string(e)` function in the user namespace.

Another question might be whether `<<` should be provided, but this issue will be handled separately (and probably won't be implemented unless there's a request).

**Comment by quicknir** [comment]
Another interesting option here: make the `to_string` a trivial template. This way, it's not instantiated until called. So the same macro could be reused for non-unique enums, an error would only occur if they tried to use the wise_enum::to_string function.

**Comment by quicknir** [comment]
Came up with a good implementation based on simply changing the implementation function for `to_string` to a template. This basically allows options 1 and 2 to be provided without any extra macros or options but just automatically based on usage. As for 3, I think I will leave this alone; it's very easy for users to write the ADL function or define a convenience macro. If there's demand I could add the convenience macro to the library.
